### ISSUE TITLE
PancakeSwap Historical Data

### ISSUE THREAD

**Issue by whittlem** [gauss314/defi#6]
I would like to retrieve the historical data for a liquidity pool on PancakeSwap. Is this possible using your defi package? If not would you mind giving me some pointers on how to do it? Is the best/easiest way to use Web3 to iterate through the smart contract event log and construct 5 minute candlesticks manually in Python?

**Comment by gauss314** [comment]
Yes, its possible get a historial TVL by protocol, read documentation, function getProtocol()

**Comment by vallarielanda26** [comment]
To retrieve historical data for a liquidity pool on PancakeSwap, consider checking if the defi package offers a built-in method to query liquidity pool events instead of iterating manually. If that’s not available, using Web3 to filter `LiquidityAdded` and `LiquidityRemoved` events from the contract could be effective. You might also want to look into the specific ABI for the PancakeSwap router to ensure you're capturing the relevant events accurately. Have you already identified the contract address and the specific events you need to listen to?
